### DISCUSSION TITLE
Cannot upgrade to laravel 10x because it cannot coexist with rebing dependency illuminate/contracts and illuminate/support

### DISCUSSION THREAD

**Discussion by drewg2009** [rebing/graphql-laravel#1021] [Q&A] [answered]
### Versions:
- graphql-laravel Version: #.#.#
- Laravel Version: #.#.# 
- PHP Version: #.#.#

### Description:
While trying to upgrade to laravel 10x, it fails because it says
```
    - rebing/graphql-laravel 8.2.1 requires illuminate/contracts ^6.0|^7.0|^8.0|^9.0 -> satisfiable by illuminate/contracts[v6.0.0, ..., 6.x-dev, v7.0.0, ..., 7.x-dev, v8.0.0, ..., 8.x-dev, v9.0.0-beta.1, ..., 9.x-dev].
    - Only one of these can be installed: illuminate/contracts[dev-master, v5.1.1, ..., 5.8.x-dev, v6.0.0, ..., 6.x-dev, v7.0.0, ..., 7.x-dev, v8.0.0, ..., 8.x-dev, v9.0.0-beta.1, ..., 9.x-dev, v10.0.0, ..., 10.x-dev], laravel/framework[v10.0.0, ..., 10.x-dev]. laravel/framework replaces illuminate/contracts and thus cannot coexist with it.
    - Root composer.json requires laravel/framework ^10.0 -> satisfiable by laravel/framework[v10.0.0, ..., 10.x-dev].
```
and rebing-graphql depends on `illuminate/contracts`

rebing-graphql likely needs to upgrade and remove the need for illuminate/contracts

### Steps To Reproduce:

Follow upgrade guide https://laravel.com/docs/10.x/upgrade in laravel project with version < 10.

- …

**Comment by mfn** [comment, marked as answer]
Please properly fill out the template next time, specifically this part:
![image](https://user-images.githubusercontent.com/87493/230103870-8664a6ba-27f1-41b0-a973-31db0947bfb7.png)

> `rebing/graphql-laravel 8.2.1 requires…`

You need to upgrade graphq-laravel first.

You are using 8.2.1, which is from 2022-01 and there was no Laravel 10 yet. You need to upgrade to [8.6.0](https://github.com/rebing/graphql-laravel/releases/tag/8.6.0) first.

**Reply by drewg2009** [reply]
Thank you for the help!

Upgrading to 8.6.0 got rid of the rebing dependency errors! 

This can be closed!
